app/controllers/HistoryController.scala

```diff
@@ -0,0 +1,60 @@
+package controllers
+
+import io.circe.syntax._
+import javax.inject.{Inject, Singleton}
+import play.api.libs.circe.Circe
+import play.api.mvc._
+import services.HistoryService
+import scala.concurrent.ExecutionContext
+import scala.util.control.NonFatal
+
+@Singleton
+class HistoryController @Inject()(cc: ControllerComponents, historyService: HistoryService)(implicit ex: ExecutionContext)
+  extends AbstractController(cc) with Circe {
+
+  def getHeader(id: String): Action[AnyContent] = Action.async {
+    historyService
+      .getHeader(id)
+      .map(header => Ok(header.asJson))
+      .recover {
+        case NonFatal(_) => BadRequest
```

Comment:
Are you sure that bad request is a proper answer here ? I think we should return NotFound in case there is no header with such id, and InternalServerError on anything else. BadRequest should be returned only if id is not valid base58 string